arch/rv32i/src/epmp.rs

```diff
@@ -667,11 +667,17 @@ impl<const MAX_AVAILABLE_REGIONS_OVER_TWO: usize> kernel::platform::mpu::MPU
                         };
                         csr::CSR.pmpconfig_set(
                             x / 2,
-                            (disable_val | cfg_val << 8) << region_shift
+                            (disable_val << region_shift)
                                 | (csr::CSR.pmpconfig_get(x / 2) & other_region_mask),
                         );
+
                         csr::CSR.pmpaddr_set(x * 2, (start) >> 2);
                         csr::CSR.pmpaddr_set((x * 2) + 1, (start + size) >> 2);
+
+                        csr::CSR.pmpconfig_set(
+                            x / 2,
+                            (cfg_val << 8) << region_shift | (csr::CSR.pmpconfig_get(x / 2)),
+                        );
```

Comment:
did I break something here in https://github.com/tock/tock/pull/2757 ?

Reply:
Yep, for me apps stopped working when using ePMP

Reply:
The problem is that we could have junk left over in the address register. As the high priority PMP configs are used for the apps they will take effect before the kernel configs (which are the lowest priority).

We have disabled the pmp regions in `disable_app_mpu()`. If we enable them before writing the address we can lock the kernel out of execute permissions. So we have to make sure we set the address first

Reply:
I added more comments here

Reply:
Ah, I understand. I wanted to make sure this breakage would not also apply for the normal pmp file, sounds like it should not. Thanks for the explanation!